examples/protonect/src/shader/stage1.fs

```diff
@@ -106,6 +104,9 @@ void main(void)
   vec2 ab1 = processMeasurementTriple(uv, P0Table1, 3, Params.ab_multiplier_per_frq.y, saturated.y);
   vec2 ab2 = processMeasurementTriple(uv, P0Table2, 6, Params.ab_multiplier_per_frq.z, saturated.z);
   
+#if MESA_BUGGY_BOOL_CMP
```

Comment:
Actually I implemented this idea first. The first issue was `#version` must be the first line of each shader file. I was not sure if the version line is intended to stay in the shader source so this. I can do it again though.

The second thing is, there are a bunch of constant parameters that can actually be reused as macros, both in shader files and in the opencl kernel, and in cuda kernel -  create a `depth_parameters.h.in` with `MEGA_BUGGY_BOOL_CMP` and embedded it as a resource, and add the rest of the parameters to it later.